dags/hivemind_etl_helpers/src/db/website/crawlee_client.py

```
@@ -0,0 +1,116 @@
+        """
+        try:
+            root = ET.fromstring(sitemap_content)
+            namespace = {"ns": "http://www.sitemaps.org/schemas/sitemap/0.9"}
+            links = []
+            for element in root.findall("ns:url/ns:loc", namespace):
+                url = element.text.strip() if element.text else None
+                if url and url.startswith(("http://", "https://")):
+                    links.append(url)
+                return links
+        except ET.ParseError as e:
+            raise ValueError(f"Invalid sitemap XML: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)